tests/test_layout.py

```diff
@@ -395,6 +395,53 @@ def test_cycle_all_with_floating_clients(hlwm, delta, floating_clients):
         hlwm.call(['cycle_all', delta])
 
 
+@pytest.mark.parametrize("delta", [1, -1])
+@pytest.mark.parametrize("clients_minimized", [
+    # every entry here is a list that determines the number
+    # of floating clients and which of those are minimized.
+    [],  # no floating clients at all
+    [True],  # only one, but it's minimized
+    [False],  # also covered by test_cycle_all_with_floating_clients
+    [False, True, False],  # one minimized in the middle
+    [True, True],
+    [False, True, True, False],
+    [True, True, True],
+    [True, False, True, False, True],
+    [False, False, True],
+    [True, False, False],
+    # multiple skips before/after wrapping between floating and tiling layer:
+    [True, True, False, False],
+    [False, False, True, True],
+])
+def test_cycle_all_skips_minimized(hlwm, delta, clients_minimized):
+    # create one tiled client and a list of floating clients,
+    # some of which are minimized.
+    tiled_client, _ = hlwm.create_client()
+    hlwm.call('rule floating=on')
+    non_minimized_clients = []
+    for minimized in clients_minimized:
+        winid, _ = hlwm.create_client()
+        hlwm.call(f'set_attr clients.{winid}.minimized {hlwm.bool(minimized)}')
+        if not minimized:
+            non_minimized_clients.append(winid)
+    if delta == -1:
+        non_minimized_clients = list(reversed(non_minimized_clients))
+    expected_winids = non_minimized_clients + [tiled_client]
+
+    assert hlwm.get_attr('clients.focus.winid') == tiled_client
+    print("expecting the clients in the order {}".format(' '.join(expected_winids)))
```

Comment:
Leftover debug print?

Reply:
I wanted to keep it because it makes sense to have this if the test case fails.

Reply:
I see! Maybe add the message as an argument to the assert statement then, e.g. assert foo == bar, "expected foo to be bar"

And also use an f-string instead of .format()?

Reply:
I want the message in the log if one of the asserts in the loop fails, so I had to add it as a message to all of the asserts in the loop. I can replace `format` by an f-string if you strongly prefer it, but I don't see the advantage of it. I think the line is perfectly readable as it is. Writing it with an f-string implies that we need to nest single and double quotes:
```python
print(f"expecting the clients in the order {' '.join(expected_winids)}")
```

Reply:
You are right, the f-string doesn't add more readability.

Should this assertion not only apply to the client.focus.winid assertion?
Adding the print statement to that assert (and maybe moving it up, before the minimized and visible asserts) should do the trick, or am I missing something?

Reply:
If you prefer it I can do it. I don't care I just want a reasonable test case and found the print statement helpful and don't really see what's so bad about.

Reply:
The PR as such is very nice and the tests are great and look good to me! Sorry for focusing so much on that tiny detail. There is nothing bad about the print statement, I just stumbled over it while reviewing, because it felt "unusual" to have this in a unit test (at least that is my feeling).

In case the tests go green, this statement would not add any value, but on the other hand will be very helpful in case of the assertion being triggered and that is why I thought combining it as an argument for the assert statement would be a good solution.

Reply:
Sorry for my frustated answer yesterday. I have no experience whether it is usual to have print statements in unit tests, if there are any reasonable arguements against such print statements I'd be interested to hear them.
For non-reproducible race conditions it is good to have as much information as possible in the log file when the race condition happens to happen. And this is why I didn't want to infer which assert statement is the one that fails.

Reply:
There is no real reason against it, just my gut feeling that it is "unusual" and "I never do it this way" ;)

I am sorry, too. This may have read like I have stronger feelings about this than I actually have.

So let's close this discussion.